src/main/java/io/github/zlika/reproducible/TarStripper.java

```diff
@@ -0,0 +1,151 @@
+    }
+
+    /**
+     * Factory that create a new instance of tar output stream, for allow extension for different file compression
+     * format.
+     *
+     *
+     * @param out
+     * @return
+     * @throws FileNotFoundException
+     *             if the file as parameter is not found
+     * @throws IOException
+     *             if there are error reading the file given as parameter
+     *
+     */
+    protected TarArchiveOutputStream createOutputStream(File out) throws FileNotFoundException, IOException
+    {
+        return new TarArchiveOutputStream(new FileOutputStream(out));
+    }
+
+    @Override
+    public void strip(File in, File out) throws IOException
+    {
+        final Path tmp = Files.createTempDirectory("tmp-" + in.getName());
+
+        List<TarArchiveEntry> sortedNames = new ArrayList<>();
+        try (final TarArchiveInputStream tar = createInputStream(in))
+        {
+
+            TarArchiveEntry entry;
+            while ((entry = tar.getNextTarEntry()) != null)
+            {
+                sortedNames.add(entry);
+                File copyTo = new File(tmp.toFile(), entry.getName());
+                if (entry.isDirectory())
+                {
+                    copyTo.mkdirs();
+                }
+                else
+                {
+                    File destParent = copyTo.getParentFile();
+                    destParent.mkdirs();
+                    Files.copy(tar, copyTo.toPath());
+                }
+            }
+        }
+        sortedNames = sortTarEntries(sortedNames);
+        try (final TarArchiveOutputStream tout = createOutputStream(out))
+        {
+            for (TarArchiveEntry entry : sortedNames)
+            {
+                File copyFrom = new File(tmp.toFile(), entry.getName());
+                if (!entry.isDirectory())
+                {
+                    final byte[] fileContent = Files.readAllBytes(copyFrom.toPath());
+                    entry.setSize(fileContent.length);
+                    tout.putArchiveEntry(filterTarEntry(entry));
+                    tout.write(fileContent);
+                    tout.closeArchiveEntry();
+                }
+                else
+                {
+                    tout.putArchiveEntry(filterTarEntry(entry));
+                    tout.closeArchiveEntry();
+                }
+            }
+        }
+        FileUtils.deleteDirectory(tmp.toFile());
+    }
+
+    private List<TarArchiveEntry> sortTarEntries(List<TarArchiveEntry> sortedNames)
+    {
+        return sortedNames.stream().sorted((a, b) -> a.getName().compareTo(b.getName()))
+                .collect(Collectors.toList());
+    }
+
+    private TarArchiveEntry filterTarEntry(TarArchiveEntry entry)
+    {
+        // Set times
+        entry.setModTime(0);
+        // Remove GID and UID that are not deterministic, to evaluate though
+        entry.setGroupId(1000);
+        entry.setUserId(1000);
```

Comment:
I think we should better set groupId and userId to 0 (i.e. root), because:
* we are sure that the "root" account exists
* according to what I have read, it seems that tar uses the current user's ID when extracting a file owned by root (to be checked)